lib/src/sdk/sdk_package_config.dart

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) 2024, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// The top level structure of an `sdk_packages.yaml` file.
+class SdkPackageConfig {
+  /// The name of the SDK this configuration is for.
+  ///
+  /// SDKs should validate the config is for them on load.
+  final String sdk;
+
+  /// All the packages vendored by this SDK.
+  final List<SdkPackage> packages;
+
+  SdkPackageConfig(this.sdk, this.packages);
+
+  // Note: yaml has `Object?` keys.
+  SdkPackageConfig.fromMap(Map<Object?, Object?> map)
+      : sdk = map['sdk'] as String,
```

Comment:
Do we need error checking here, or we just expect the sdk to always have correctly structured data?

Reply:
I think it is pretty safe to assume correctly structured data, but it could help SDK engineers if we had better error messages. I think it should still throw either way.

Reply:
Added a todo (and a version)